public/app/features/search/hooks/useSearchQuery.ts

```diff
@@ -23,6 +24,9 @@ const updateLocation = debounce((query) => locationService.partial(query), 300);
 export const useSearchQuery = (defaults: Partial<DashboardQuery>) => {
   const queryParams = parseRouteParams(locationService.getSearchObject());
   const initialState = { ...defaultQuery, ...defaults, ...queryParams };
+  if (!initialState.layout?.length) {
+    initialState.layout = localStorage.getItem(SEARCH_SELECTED_LAYOUT) as SearchLayout;
+  }
```

Comment:
```suggestion
  if (!queryParams.layout?.length) {
    initialState.layout = localStorage.getItem(SEARCH_SELECTED_LAYOUT) as SearchLayout;
  }
```

to avoid using `default.layout`?

Reply:
maybe -- it breaks a test I did not want to look into :) and works for the cases we care about

Reply:
hmmm, is this the intended behavior? I'd expect it `grid` to remain the selected layout throughout this flow

https://user-images.githubusercontent.com/23451458/173638640-6a0e39ca-6b35-48dd-9d9f-5f4462302c71.mp4

Reply:
Maybe better:  
![2022-06-14_14-32-12 (1)](https://user-images.githubusercontent.com/705951/173692884-ca4616da-24b1-41be-905d-36afe4408f50.gif)

It still has some issues when you switch between the views and the component is already loaded (i think we have to live with that, or figure out how somehow make sure the manage view completely re-initializes after loading/closing the search overlay)